tests/v2/validation/rbac/rbac.go

```diff
@@ -84,3 +100,61 @@ func createProject(client *rancher.Client, clusterID string) (createProject *man
 	return createProject, err
 
 }
+
+func getPSALabels(response *v1.SteveAPIObject, actualLabels map[string]string) map[string]string {
+	expectedLabels := map[string]string{}
+
+	for label := range response.Labels {
+		if _, found := actualLabels[label]; found {
+			expectedLabels[label] = actualLabels[label]
+		}
+	}
+	return expectedLabels
+
+}
+
+func createDeployment(steveclient *v1.Client, client *rancher.Client, clusterID string, containerName string, image string, namespaceName string) (*v1.SteveAPIObject, error) {
+
+	deploymentName := namegen.AppendRandomString("")
+	containerTemplate := workloads.NewContainer(containerName, image, coreV1.PullAlways, []coreV1.VolumeMount{}, []coreV1.EnvFromSource{})
+	matchLabels := map[string]string{}
+	matchLabels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.deployment-%v-%v", namespaceName, deploymentName)
+
+	podTemplate := workloads.NewPodTemplate([]coreV1.Container{containerTemplate}, []coreV1.Volume{}, []coreV1.LocalObjectReference{}, matchLabels)
+	deployment := workloads.NewDeploymentTemplate(deploymentName, namespaceName, podTemplate, matchLabels)
+
+	deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+	err = kwait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		steveclient, err := client.Steve.ProxyDownstream(clusterID)
+		if err != nil {
+			return false, nil
+		}
```

Comment:
Doesn't this function already use the client argument as proxied to the downstream?

In my opinion, either this part needs a removal or proxied steve client instantiation could happen as the first thing in this helper function after L116.

Reply:
yes, good point. Removed the statement where I instantiate steveclient